filebeat/fileset/factory.go

```diff
@@ -130,7 +134,7 @@ func (p *inputsRunner) Start() {
 		callback := func(esClient *elasticsearch.Client) error {
```

Comment:
Reading the code on line 124 and 129: In case of an error, should we register a callback at all? Not introduced in this PR but probably something we should clean up.

Reply:
I think this code is correct. The lines you are referring to are trying to load pipelines immediately in `Start`. It's possible it fails due to e.g ES is not being not available.
But it's the whole point of using OnConnect callbacks. If previously pipeline loading fails, it can be attempted again when connecting to ES again.

Reply:
@kvch I got tricked by this as well. Can you add a comment why the callback must always be registered:
- if pipeline loading fails, retry on reload
- always retry on reload in case of someone removing it by accident or new ES cluster in place